content/testing.md

```diff
@@ -0,0 +1,256 @@
+
+Finally you may wish to test that your application works under typical load or see how much load it can handle before it falls over. This is called a *load test* or *stress test*. Such tests can be challenging to set up and typically aren't run often but are very important for confidence before a big production launch.
+
+<h3 id="challenges-with-meteor">Challenges of testing in Meteor</h3>
+
+2. Challenges of testing a Meteor application
+  1. The client/server divide, ensuring data is available on the client
+  2. Reactivity + testing the system responds to changing data properly
+
+
+<h2 id="test-modes">Test modes in Meteor</h2>
+
+- Various ways to run your application that allows different testing modalities
+
+- `meteor test` simply loads your test files and allows you to import modules you want to test
+
+- `meteor test --full-app` runs your app as usual with extra test files included
+
+- `meteor test-packages` is a way of testing Atmosphere packages, which we'll discuss in more detail in the [Writing Packages Article](writing-packages.html).
+
+<h3 id="driver-packages">Driver packages</h3>
+
+A test driver is a mini-application that runs in place of your app and runs each of your defined tests (each of which `import` relevant sections of your app), whilst reporting the results in some kind of user interface.
+
+There are two main kinds of test driver packages:
+  -web-reporters which are Meteor applications and display a special test reporting web UI that you can view the test results in [include a SS]
+
+  - console-reporters that run completely on the command-line and are primary used for automated testing like [continuous integration](#ci) (as we'll see, typically PhantomJS is used to drive such tests).
+
+While developing your app, chances are you'll want to run unit tests against a web reporter; for our example we will use a reporter that renders to Mocha's default web UI. We can add the driver simply by adding the [`avital:mocha`](https://atmospherejs.com/avital/mocha) package to our app.
+
+```bash
+meteor add avital:mocha
+```
+
+This package also doesn't do anything in development or production mode, but when our app is run in [test](#test-mode) or [full-app](#full-app-test-mode) test mode, it takes over, running test code on both the client and server, and rendering results to the browser.
+
+In this article, we'll use the popular [Mocha](https://mochajs.org) test runner alongside the [Chai](http://chaijs.com) assertion library to test our application. In order to write tests in Mocha, we can add the [`avital:mocha`](https://atmospherejs.com/avital/mocha) package to our app.
+
+```
+meteor add avital:mocha
+```
+
+<h3 id="testing-externally">Testing externally</h3>
```

Comment:
I feel like end-to-end tests is a better name? I don't really know what "external" means for tests